openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -425,27 +424,41 @@ public static OpenAPI getOpenAPIFromOpenAPIV3Parser(Path definitionPath) throws
         return parseResult.getOpenAPI();
     }
 
+    /**
+     * Check whether the given media type is currently supported in the tool.
+     *
+     * @param mediaTypeEntry
+     * @return
+     */
+    public static boolean isSupportedMediaType(Map.Entry<String,
+            io.swagger.v3.oas.models.media.MediaType> mediaTypeEntry) {
+        String mediaType = mediaTypeEntry.getKey();
+        String defaultBallerinaType = getBallerinaMediaType(mediaType, true);
+        if (defaultBallerinaType.equals(HTTP_REQUEST) &&
+                !(mediaTypeEntry.getValue().getSchema() != null && mediaType.equals(MediaType.MULTIPART_FORM_DATA))) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
     /**
      * Generate BallerinaMediaType for all the return mediaTypes.
      */
     public static String getBallerinaMediaType(String mediaType, boolean isRequest) {
-
-        switch (mediaType) {
-            case MediaType.APPLICATION_JSON:
-                return SyntaxKind.JSON_KEYWORD.stringValue();
-            case MediaType.APPLICATION_XML:
-                return SyntaxKind.XML_KEYWORD.stringValue();
-            case MediaType.APPLICATION_FORM_URLENCODED:
-            case MediaType.TEXT_HTML:
-            case MediaType.TEXT_PLAIN:
-                return STRING_KEYWORD.stringValue();
-            case IMAGE_PNG:
-            case MediaType.APPLICATION_OCTET_STREAM:
-            case APPLICATION_PDF:
-            case ANY_TYPE:
-                return isRequest ? HTTP_REQUEST : HTTP_RESPONSE;
-            default:
-                return SyntaxKind.JSON_KEYWORD.stringValue();
+        if (mediaType.equals(MediaType.APPLICATION_JSON) || mediaType.matches("application/.*\\+json") ||
+                mediaType.matches(".*/json")) {
```

Comment:
Whats the purpose of having `mediaType.matches(".*/json")` in L450? doesn't it get covered by the previous 2 conditions?

Reply:
It is to cover the scenarios like `text/json`. The common case is `application/json`. But we can remove the `application/json` check and keep only the `mediaType.matches(".*/json")` check. Is it fine ?

Reply:
my bad I didn't notice `/` character in the regex. +1 to keep and remove the `application/json` check as it becomes redundant with the last condition